File: src/main/java/edu/metrostate/trackingsystem/application/services/VehicleService.java
//This class is used for transferring data through application layers from the database
package edu.metrostate.trackingsystem.application.services;

import edu.metrostate.trackingsystem.domain.models.Vehicle;
import edu.metrostate.trackingsystem.domain.repositories.VehicleRepository;
import edu.metrostate.trackingsystem.infrastructure.database.Result;
import javafx.collections.ObservableList;

public class VehicleService {

    private final VehicleRepository repository;

    public VehicleService(VehicleRepository repository) {
        this.repository = repository;
    }

    public Result<Boolean> deleteVehicle(String id, String dealerId) {
        return repository.deleteVehicle(id, dealerId);
    }

    public Result<Boolean> addVehicle(Vehicle vehicle) {
        return repository.addVehicle(vehicle);
    }

    public ObservableList<Vehicle> getVehicles() {
        return repository.getVehicles();
    }

    public void toggleIsRented(Vehicle vehicle) { repository.toggleIsRented(vehicle);}
}

File: src/main/kotlin/edu/metrostate/dealership/application/services/VehicleService.kt
//This class is used for transferring data through application layers from the database
package edu.metrostate.dealership.application.services

import edu.metrostate.dealership.domain.models.Vehicle
import edu.metrostate.dealership.domain.repositories.VehicleRepository
import edu.metrostate.dealership.infrastructure.database.Result
import javafx.collections.ObservableList

class VehicleService(private val repository: VehicleRepository) {
    fun deleteVehicle(id: String): Boolean {
       return repository.deleteVehicle(id)
    }

    fun addVehicle(vehicle: Vehicle): Result<Boolean> {
        return repository.addVehicle(vehicle)
    }

    val vehicles: ObservableList<Vehicle>
        get() = repository.vehicles

    fun toggleIsRented(vehicleId: String) {
        repository.toggleIsRented(vehicleId)
    }
}